### PR TITLE
UTS39 C3

### PR DESCRIPTION
**Pull request by josevalim**
Follow up to #11620.

### CODE REVIEW

File: lib/mix/test/mix/tasks/release_test.exs [hunk 1]
```diff
@@ -330,7 +330,7 @@ defmodule Mix.Tasks.ReleaseTest do
         assert %{
                  app_dir: app_dir,
                  cookie_env: ^cookie,
-                 encoding: {:time_μs, :"£", "£", '£'},
+                 encoding: {:_μ, :"£", "£", '£'},
```
Comment (mrluc) [inline lib/mix/test/mix/tasks/release_test.exs:333]: Ah -- so it resolves to {Greek} this way! Sweet.

**Comment by josevalim** [comment]
Hi @eksperimental, this is the last PR related to Unicode Security. Therefore the list of updated documents can be found in String.Unicode. Thank you! ❤️

File: lib/elixir/pages/unicode-syntax.md [hunk 2]
```diff
@@ -2,15 +2,54 @@
...
+
+The characters allowed in identifiers are the ones specified by Unicode. Generally speaking, it is restricted to characters typically used by the writing system of human languages still in activity. In particular, it excludes symbols such as emojis, alternate numeric representations, musical notes, etc.
+
+Elixir imposes many restrictions on identifiers for security purposes. For example, the word "josé" can be written in two ways in Unicode: as the combination of the characters `j o s é` and as a combination of the characters `j o s e ́ `, where the accent is its own character. The former is called NFC form and the latter is the NFD form. Elixir requires all characters to be the in the NFC form.
+
+Elixir also disallows mixed-scripts in most scenarios. For example, it is not possible to name a variable `аdmin`, where `а` is in Cyrillic and the remaining characters are in Latin. Doing so, will raise the following error:
+
+```elixir
```
Comment (josevalim) [inline lib/elixir/pages/unicode-syntax.md:21]:
```suggestion
```
```

**Comment by eksperimental** [comment]
> Hi @eksperimental, this is the last PR related to Unicode Security. Therefore the list of updated documents can be found in String.Unicode. Thank you! heart

Great, I will be updating the job for the Maintenance app soon.

**Comment by josevalim** [comment]
@mrluc I am done with the merging. I have also merged the Unicode Syntax and Unicode Security documents into a single one, since they are very related now as we don't allow syntax that violates C1 and C3. If you could do one final pass on the docs and the docs, I would appreciate it very much!

File: lib/elixir/unicode/tokenizer.ex [hunk 3]
```diff
@@ -112,137 +106,362 @@ defmodule String.Tokenizer do
...
+  unicode_continue = Enum.map(unicode_continue, &{&1, Map.get(codepoints_to_mask, &1, top)})
+
+  rangify = fn [{head, scriptset} | tail] ->
+    {first, last, scriptset, acc} =
+      Enum.reduce(tail, {head, head, scriptset, []}, fn
+        {number, scriptset}, {first, last, scriptset, acc} when number == first - 1 ->
+          {number, last, scriptset, acc}
+
+        {number, scriptset}, {first, last, range_scriptset, acc} ->
+          {number, number, scriptset, [{first, last, range_scriptset} | acc]}
```
Comment (mrluc) [inline lib/elixir/unicode/tokenizer.ex:280]: On a first read, I worried that there was a risk this wouldn't work at boundaries where one range is in one scriptset, and the other is in another scriptset. But looking closer, I see I was overlooking that `scriptset` is matched with the accumulated/current one in that clause 👍

File: lib/elixir/unicode/tokenizer.ex [hunk 4]
```diff
@@ -112,137 +106,362 @@ defmodule String.Tokenizer do
...
+      defp unicode_upper(unquote(first)), do: unquote(scriptset)
     else
-      defp unicode_start?(entry) when entry in unquote(first)..unquote(last), do: true
+      defp unicode_upper(entry) when entry in unquote(first)..unquote(last),
+        do: unquote(scriptset)
     end
   end
 
-  defp unicode_start?(_), do: false
+  defp unicode_upper(_), do: 0
```
Comment (mrluc) [inline lib/elixir/unicode/tokenizer.ex:295]: Same question a bunch of places:

There's a mix of functions that use `0` directly, and those that use `@bottom` -- if we're going to get `@bottom` from  `lattice(size)` then should we use that everywhere? I don't know if module attrs have a runtime hit though or if they're compiled in.

Comment (mrluc) [reply lib/elixir/unicode/tokenizer.ex:295]: Looks like they're compiled in, from the terminology used in `h Kernel.@` ('takes a snapshot' among other phrases), so should be safe to change those all to `@bottom`; I tested doing that locally and tests still passed, but I can't push to this branch. Very low-importance change obviously.

Comment (josevalim) [reply lib/elixir/unicode/tokenizer.ex:295]: Oh, I was considered them to be unrelated “zeros” but treating it as a `@bottom` makes semantic sense too, as no codepoint should have no script set.

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Comment by PUNTO-HIDALGO** [comment]
@bruce uts39-c3

**Comment by eksperimental** [comment]
> Hi @eksperimental, this is the last PR related to Unicode Security. Therefore the list of updated documents can be found in String.Unicode. Thank you! heart

@josevalim Just to inform that the Maintenance has been updated with the changes up to today related to Unicode
The update is here https://github.com/eksperimental/maintenance/commit/f752db257120bb493a0d640cd812e9abd8740ca9
and the app is running with this version.
So any new Unicode version should be picked up within hours.

**Comment by josevalim** [comment]
Perfect, thank you @eksperimental!

**Pull request merged**